Note that this patch is also applied to the host toolchain
in termux_setup_golang().

diff -u -r ../go/src/os/exec_posix.go ./src/os/exec_posix.go
--- ../go/src/os/exec_posix.go	2024-03-29 15:27:02.000000000 +0000
+++ ./src/os/exec_posix.go	2024-05-24 22:23:22.855862054 +0000
@@ -51,6 +51,16 @@
 		sysattr.Files = append(sysattr.Files, f.Fd())
 	}
 
+	if runtime.GOOS == "android" {
+		argv = append([]string{argv[0]}, argv...)
+		argv[1] = name
+		if runtime.GOARCH == "arm" || runtime.GOARCH == "386" {
+			name = "/system/bin/linker"
+		} else {
+			name = "/system/bin/linker64"
+		}
+		}
+
 	pid, h, e := syscall.StartProcess(name, argv, sysattr)
 
 	// Make sure we don't run the finalizers of attr.Files.
